pkg/archive/paths.go

```diff
@@ -37,25 +35,28 @@ const (
 
 var (
 	extensions = map[string]FormatType{
-		".ear":     ZipArchive,
-		".jar":     ZipArchive,
-		".par":     ZipArchive,
-		".war":     ZipArchive,
-		".zip":     ZipArchive,
-		".tar":     TarArchive,
-		".tar.gz":  TarGzArchive,
-		".tgz":     TarGzArchive,
-		".tar.bz2": TarBz2Archive,
-		".tbz2":    TarBz2Archive,
+		"ear":     ZipArchive,
+		"jar":     ZipArchive,
+		"par":     ZipArchive,
+		"war":     ZipArchive,
+		"zip":     ZipArchive,
+		"tar":     TarArchive,
+		"tar.gz":  TarGzArchive,
+		"tgz":     TarGzArchive,
+		"tar.bz2": TarBz2Archive,
+		"tbz2":    TarBz2Archive,
 	}
 )
 
-// A WalkFn iterates through an archive, calling FileWalkFn on each member file.
-type WalkFn func(ctx context.Context, path string, walkFn FileWalkFn) error
+// A WalkZipFn and WalkTarFn iterates through an archive, calling FileWalkFn on each member file.
+type WalkZipFn func(ctx context.Context, path string, walkFn FileWalkFn) error
```

Comment:
No need to rename this, it's not specific to zips, it just happens to be used for zips in our codebase at present..
This may disappear if the recursion PR gets merged anyway, though.